src/Symfony/Component/Lock/Store/ZookeeperStore.php

```diff
@@ -0,0 +1,183 @@
+     */
+    public function save(Key $key)
+    {
+        if ($this->exists($key)) {
+            return;
+        }
+
+        $resource = $this->getKeyResource($key);
+        $token = $this->getUniqueToken($key);
+
+        try {
+            $this->createNewLock($resource, $token);
+        } catch (LockConflictedException $ex) {
+            throw $ex;
+        } catch (\Throwable $throwable) {
+            throw new LockAcquiringException($throwable);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws LockReleasingException
+     */
+    public function delete(Key $key)
+    {
+        if (!$this->exists($key)) {
+            return;
+        }
+        $resource = $this->getKeyResource($key);
+        try {
+            $this->zookeeper->delete($resource);
+        } catch (\Throwable $throwable) {
+            // For Zookeeper Ephemeral Nodes, the node will be deleted upon session death. But, if we want to unlock
+            // the lock before proceeding further in the session, the client should be aware of this
+            throw new LockReleasingException($throwable);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function exists(Key $key): bool
+    {
+        $resource = $this->getKeyResource($key);
+        try {
+            return $this->zookeeper->get($resource) === $this->getUniqueToken($key);
+        } catch (\ZookeeperException $ex) {
+            return false;
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws NotSupportedException
+     */
+    public function waitAndSave(Key $key)
+    {
+        throw new NotSupportedException();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function putOffExpiration(Key $key, $ttl)
+    {
+        // This store doesn't support TTL. The lock nodes die at the end of a client's session if not released correctly
+    }
+
+    /**
+     * Create a zookeeper node.
+     *
+     * @param string $node  The node which needs to be created
+     * @param string $value The value to be assigned to a zookeeper node
+     *
+     * @throws LockConflictedException
+     * @throws LockAcquiringException
+     */
+    private function createNewLock(
+        string $node,
+        string $value
+    ) {
+        // Default Node Permissions
+        $acl = array(array('perms' => \Zookeeper::PERM_ALL, 'scheme' => 'world', 'id' => 'anyone'));
+        // This ensures that the nodes are deleted when the client session to zookeeper server ends.
+        $type = \Zookeeper::EPHEMERAL;
+
+        try {
+            $this->zookeeper->create($node, $value, $acl, $type);
+        } catch (\ZookeeperException $ex) {
+            if (\Zookeeper::NODEEXISTS === $ex->getCode()) {
+                throw new LockConflictedException($ex);
+            }
+
+            throw new LockAcquiringException($ex);
+        }
+    }
+
+    /**
+     * @param Key $key Key for the lock
+     *
+     * @return string
```

Comment:
I'd suggest to remove the docblock entirely, it doesn't provide much over the signature